src/src/com/microsoft/aad/adal/Oauth2.java

```diff
@@ -516,12 +517,12 @@ private AuthenticationResult postMessage(String requestMessage, HashMap<String,
 
                 Logger.v(TAG, "Server error message:" + errMessage);
                 if (response.getResponseException() != null) {
-                    throw response.getResponseException();
+                    throw new IOException(response.getResponseException());
```

Comment:
If the only exceptions that are going to be returned from getResponseException are IOExceptions, why wrap it in another IOException before rethrowing it?